components/document-renderer.tsx

```diff
@@ -8,16 +9,30 @@ import { PageRenderer } from "./page-renderer";
  * Each of these cases map to a `section` found in the `.tina/settings.yml`
  *
  */
-export const DocumentRenderer = (props: Tina.SectionDocumentUnion) => {
-  switch (props.__typename) {
-    case "Pages_Document":
-      return <PageRenderer {...props.data} />;
+
+export const DocumentRenderer = ({
+  section,
+  document,
+}: {
+  section: String;
+  document: Tina.SectionDocumentUnion;
+}) => {
+  switch (section) {
+    case "shelves":
+      return <ShelvesRenderer {...document.data} />;
+    case "items":
+      return <ItemsRenderer {...document.data} />;
     default:
-      return <NoData />;
+      return <NoData section={section} />;
   }
 };
 
-const NoData = () => {
-  console.error("Woops, this shouldn't be rendered!");
-  return <pre>No data</pre>;
+const NoData = ({ section }: { section: String }) => {
+  return (
+    <pre>
+      [Error] Renderer for section:"{section}" is missing. Consider adding a{" "}
+      {section[0].toUpperCase() + section.slice(1)}Renderer Component to
+      DocumentRenderer?
+    </pre>
+  );
```

Comment:
As I swapped out the `pages` section for my own, I noticed I was getting the error: "Woops, this shouldn't be rendered!"

I realized that I was seeing this error because I had not _yet_ created a `Renderer` for my newly added sections.  I decided to make a more clearer error message (with a suggestion).

In fact, `No data` is not even correct, is it?  There could be data returned from the `query`, but just no `Renderer` to display it.

Reply:
Agreed, this is a better message